src/internal/storage/chunk/batcher.go

```diff
@@ -0,0 +1,133 @@
+package chunk
+
+import (
+	"context"
+
+	"golang.org/x/sync/semaphore"
+)
+
+// ChunkFunc is a function that provides the metadata for the entries in a chunk and a data reference to the chunk.
+type ChunkFunc = func([]interface{}, *DataRef) error
+
+// EntryFunc is a function that provides the metadata for an entry and a data reference to the entry in a chunk.
+// Size zero entries will have a nil data reference.
+type EntryFunc = func(interface{}, *DataRef) error
+
```

Comment:
did I miss somewhere where the metadata isn't `*index.Index/nil` or is this just for flexibility?

Reply:
It is primarily for flexibility, but it would also be a circular dependency if it was typed to `*index.Index`.